app/models/cart.rb

```diff
@@ -0,0 +1,16 @@
+class Cart
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || {}
+  end
+
+  def add_beer(beer, quantity)
+    if contents.has_key?(beer.id)
+      contents[beer.id][:quantity] += quantity
+      contents[beer.id][:beer_price] = beer.price * contents[beer.id][:quantity]
+    else
+      contents[beer.id] = {beer_name: beer.name, beer_price: beer.price, quantity: quantity} 
```

Comment:
Line is too long. [93/80]<br>Space inside { missing.<br>Space inside } missing.<br>Trailing whitespace detected.